src/pages/SearchBar/SearchBar.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import Button from '../../components/Button/Button';
+import ButtonInactive from '../../components/ButtonInactive/ButtonInactive';
+import {
+  SearchBarForm,
+  Heading,
+  SearchBarContainer,
+  Prefix,
+  Input,
+} from './SearchBar.style';
+
+function SearchBar(props) {
+  // Get the subreddit value from the URL
+  const [searchTerm, setSearchTerm] = useState((new URLSearchParams(window.location.search).get('subreddit')).slice(0, 21));
+  const [isValidSearch, setIsValidSearch] = useState(true);
+  const { handleSearch } = props;
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    if (isValidSearch) {
+      handleSearch(searchTerm);
+    }
+  }
+
+  function handleChange(event) {
+    // Check if last letter of input is valid
+    const { value } = event.currentTarget;
+    const regexValidInput = /\w+/;
+    const isValidInput = regexValidInput.test(value[value.length - 1]);
+    if (isValidInput) {
+      setSearchTerm(value);
+    }
+  }
+
+  useEffect(() => {
+    // Check the entire query string is valid, used for binary states
+    const regexValidQuery = /^[a-z]{1}[\w]{2,20}$/gmi;
```

Comment:
I have added comments in the code regarding the two regex expressions.

I spent quite a bit of time learning regex after I had some trouble with it and try to use it now where I can. I can change it to get validated through pure JS if you prefer. Here are the explanations:

**/\w+/**
The first regex expression `/\w+/` is very basic and just ensures that the input is either alphanumerical or an underscore. I'm checking the very last letter of the input instead of parsing the whole string (which I would usually do) as it seems like a waste to check the whole input value.

**/^[a-z]{1}[\w]{2,20}$/**

This expression might seem complicated at first but it's very simple. The `\w` value in Regex is for any letter, number or underscore.

It will require that:
1. First character must be a letter
2. Characters 1 to 20 must be alphanumerical or an underscore.

`^`: Start the parse from beginning of the string
`[a-z]{1}`: Ensure that the first char is a letter
`[\w]{2,20}`: Allow between 2 and 20 alphanumerical and underscores
`$`: Must parse the whole string/to end of string

I have left the Regex in for now, but I can turn it into pure JS checks if you would prefer that.

Reply:
No need for pure JS checks. I'd just ensure that there is an explanation. It's important to understand what the intention of the code is. There might be a bug in the RegExp which is hard to find if you don't know what it's supposed to do. Since you added a comment that's fine